integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/sort/SearchSortIT.java

```diff
@@ -183,39 +176,37 @@ public void separateSort() {
 				.toQuery();
 		assertThat( query )
 				.hasDocRefHitsExactOrder( INDEX_NAME, THIRD_ID, SECOND_ID, FIRST_ID, EMPTY_ID );
-
-		query = scope.query()
-				.predicate( f -> f.matchAll() )
-				.sort( f -> f.by( sortDesc ) )
-				.toQuery();
-		assertThat( query )
-				.hasDocRefHitsExactOrder( INDEX_NAME, THIRD_ID, SECOND_ID, FIRST_ID, EMPTY_ID );
 	}
 
 	@Test
 	public void lambda_caching() {
 		AtomicReference<SearchSort> cache = new AtomicReference<>();
 
-		Function<? super SearchSortFactoryContext, ? extends SearchSortTerminalContext> cachingContributor = c -> {
+		Function<? super SearchSortFactoryContext, ? extends SearchSort> cachingContributor = c -> {
 			SearchSort result = cache.get();
 			if ( cache.get() == null ) {
 				result = c.byField( "string" ).onMissingValue().sortLast().toSort();
 				cache.set( result );
 			}
-			return c.by( result );
+			return result;
 		};
 
 		Assertions.assertThat( cache ).hasValue( null );
 
+		StubMappingScope scope = indexManager.createScope();
 		SearchQuery<DocumentReference> query;
 
-		query = simpleQuery( cachingContributor );
+		query = scope.query().predicate( f -> f.matchAll() )
+				.sort( cachingContributor.apply( scope.sort() ) )
+				.toQuery();
```

Comment:
Probably it's not relative do this pull request.

But I think that with this code we're not actually testing the caching of a `SearchSort` by Hibernate Search.
It looks like we're testing our `cachingContributor' function instead.

I believe that reuse the same `SearchSort` two times would be the same, but simpler.

Reply:
Since caching is in an internal behavior. We could test the reuse of an instance, not the caching.

Reply:
You're right. This used to make sense when we used "caching lambdas", but it doesn't make much sense anymore.

When you're done adding more relevant tests in your next PR, maybe you can remove that test?

Reply:
Yep! Let's merge it. I'll handle it in [HSEARCH-3192](https://hibernate.atlassian.net/browse/HSEARCH-3192).